modules/ejbca-common/src/org/ejbca/core/model/ca/publisher/LdapPublisher.java

```diff
@@ -653,27 +665,52 @@ public void revokeCertificate(AuthenticationToken admin, Certificate cert, Strin
 			}
 			if (oldEntry != null) {          
 				if (removecert) {
-					// Don't try to remove the cert if there does not exist any
-					LDAPAttribute oldAttr = oldEntry.getAttribute(getUserCertAttribute());
-					if (oldAttr != null) {
-						modSet = getModificationSet(oldEntry, certdn, null, false, true, null, cert);
-						LDAPAttribute attr = new LDAPAttribute(getUserCertAttribute());
-						modSet.add(new LDAPModification(LDAPModification.DELETE, attr));                    
-					} else {
-						String msg = intres.getLocalizedMessage("publisher.inforevokenocert");
-						log.info(msg);
-					}            		
+                    // Get the current set of certificates
+                    LDAPAttribute oldAttr = oldEntry.getAttribute(getUserCertAttribute());
+                    if (oldAttr != null) {
+                        modSet = getModificationSet(oldEntry, certdn, null, false, true, null, cert);
+                        // Remove the revoked cert from the attribute
+                        try {
+                            oldAttr.removeValue(cert.getEncoded());
+                        } catch (CertificateEncodingException e) {
+                            // Exception should not happen!
+                            String msg = "Unexpected certificate encoding issue. Cannot remove the certificate from LDAP.";
+                            log.error(msg);
```

Comment:
In storeCertificate the same encoding error throws a PublisherException. It would be consistent to do that here as well I think.